### ISSUE TITLE
Support RNNs and LSTMs

### ISSUE THREAD

**Issue by shradhab** [tensorflow/tensorflow#15845]
Feature Request for TF Lite - support for the following:
td_split
tf_pad (Padding is something that an RNN needs de-facto unless tensorflow RNNs accept lists of variously shaped tensors.)
tf_gather
tf_slice
LSTM Cells
Dynamic RNNs
Are there alternatives to the above in TF Lite?

Have I written custom code - No
OS Platform and Distribution - Android
TensorFlow installed from - https://github.com/tensorflow
TensorFlow version - 1.4
Bazel version - 0.5.4
CUDA/cuDNN version - NA
GPU model and memory - Na
Exact command to reproduce - NA

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by shradhab** [comment]
Updated.

--
Shradha

**Comment by andrehentz** [comment]
@shradhab Thanks! Support for gather and pad will come soon, with split and possibly slice following right after. We do have op for LSTMs and RNNs in the interpreter, but the conversion step still needs some careful work.

**Comment by shradhab** [comment]
Thanks! Will TF Lite automatically target Qualcomm DSP speed up or will it still require a separate binary?

**Comment by andrehentz** [comment]
The plan is to have one binary that can work with or without hardware acceleration.

**Comment by tensorflowbutler** [comment]
The original poster has replied to this issue after the stat:awaiting response label was applied.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by RoboEvangelist** [comment]
are there any pretrained RNN/LSTM models in googl's model zoo? I cannot find any google links regarding that matter

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by RoboEvangelist** [comment]
Seems like somebody figured out how to convert a specific LSTM model. I am trying to test, but my computer is taking a while to build toco:
https://github.com/tensorflow/tensorflow/issues/15805

Maybe you guys can also try it

**Comment by RoboEvangelist** [comment]
I was also able to convert the LSTM graph in FLOAT. I just cannot quantize to UINT8 because Unpack operation is not supported by TOCO

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @aselle, @andrehentz: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by achowdhery** [comment]
Please take a look at the recent example we added for Unidirectional LSTM test case. It may help your use case:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/experimental/examples/lstm/unidirectional_sequence_lstm_test.py

**Comment by renjie-liu** [comment]
Looks like it's a very old issue. we have some RNN/LSTMs support already,

please see here for more examples: 
https://www.tensorflow.org/lite/convert/rnn

**Comment by Saduf2019** [comment]
@shradhab 
please update as per above comment,

**Comment by shradhab** [comment]
Support available
